src/python/grpcio_test/grpc_test/_core_over_links_base_interface_test.py

```diff
@@ -0,0 +1,177 @@
+_SERVICE_TERMINAL_METADATA = ((b'6', b'stu'), (b'7', b'vwx'), (b'8', b'yza'),)
+_CODE = _intermediary_low.Code.OK
+_MESSAGE = b'test message'
+
+
+class _SerializationBehaviors(
+    collections.namedtuple(
+        '_SerializationBehaviors',
+        ('request_serializers', 'request_deserializers', 'response_serializers',
+         'response_deserializers',))):
+  pass
+
+
+class _Links(
+    collections.namedtuple(
+        '_Links',
+        ('invocation_end_link', 'invocation_grpc_link', 'service_grpc_link',
+         'service_end_link'))):
+  pass
+
+
+def _serialization_behaviors_from_serializations(serializations):
+  request_serializers = {}
+  request_deserializers = {}
+  response_serializers = {}
+  response_deserializers = {}
+  for (group, method), serialization in serializations.iteritems():
+    request_serializers[group, method] = serialization.serialize_request
+    request_deserializers[group, method] = serialization.deserialize_request
+    response_serializers[group, method] = serialization.serialize_response
+    response_deserializers[group, method] = serialization.deserialize_response
+  return _SerializationBehaviors(
+      request_serializers, request_deserializers, response_serializers,
+      response_deserializers)
+
+
+class _Implementation(test_interfaces.Implementation):
+
+  def instantiate(self, serializations, servicer):
+    serialization_behaviors = _serialization_behaviors_from_serializations(
+        serializations)
+    invocation_end_link = implementations.invocation_end_link()
+    service_end_link = implementations.service_end_link(
+        servicer, test_constants.DEFAULT_TIMEOUT,
+        test_constants.MAXIMUM_TIMEOUT)
+    service_grpc_link = service.service_link(
+        serialization_behaviors.request_deserializers,
+        serialization_behaviors.response_serializers)
+    port = service_grpc_link.add_port(0, None)
+    channel = _intermediary_low.Channel('localhost:%d' % port, None)
+    invocation_grpc_link = invocation.invocation_link(
+        channel, b'localhost',
+        serialization_behaviors.request_serializers,
+        serialization_behaviors.response_deserializers)
+
+    invocation_left_logging_link, invocation_right_logging_link = test_utilities.logging_links('Invocation')
+    service_left_logging_link, service_right_logging_link = test_utilities.logging_links('Service')
+    invocation_end_link.join_link(invocation_left_logging_link)
+    invocation_left_logging_link.join_link(invocation_end_link)
+    invocation_grpc_link.join_link(invocation_right_logging_link)
+    invocation_right_logging_link.join_link(invocation_grpc_link)
+    service_grpc_link.join_link(service_left_logging_link)
+    service_left_logging_link.join_link(service_grpc_link)
+    service_end_link.join_link(service_right_logging_link)
+    service_right_logging_link.join_link(service_end_link)
+
+    #invocation_end_link.join_link(invocation_grpc_link)
```

Comment:
Actually, because I have made a responsible calculation of how much I'm allowed to write to the log in a test, and because these could potentially write a lot, I've taken out the use of the logging links for this commit.